tests/tests_geomstats/test_discrete_curves.py

```diff
@@ -618,63 +827,511 @@ def test_space_derivative(self):
+        expected = curves
+
+        self.assertAllClose(result, expected)
+
+
+class TestClosedDiscreteCurves(ManifoldTestCase, metaclass=Parametrizer):
+    # closed discrete curves doesn't have random point
+    space = ClosedDiscreteCurves
+    skip_test_projection_belongs = True
+    skip_test_random_point_belongs = True
+    skip_test_random_tangent_vec_is_tangent = True
+    skip_test_to_tangent_is_tangent = True
+
+    class DiscreteCurvesTestData(_ManifoldTestData):
+        s2 = Hypersphere(dim=2)
+        r2 = Euclidean(dim=2)
+        r3 = Euclidean(dim=3)
+        space_args_list = [(r2,), (r3,)]
+        shape_list = [(10, 2), (10, 3)]
+        n_samples_list = random.sample(range(2, 5), 2)
+        n_points_list = random.sample(range(2, 5), 2)
+        n_vecs_list = random.sample(range(2, 5), 2)
+
+        def random_point_belongs_test_data(self):
+            smoke_space_args_list = [(self.s2,), (self.r2,)]
+            smoke_n_points_list = [1, 2]
+            return self._random_point_belongs_test_data(
+                smoke_space_args_list,
+                smoke_n_points_list,
+                self.space_args_list,
+                self.n_points_list,
+            )
+
+        def projection_belongs_test_data(self):
+            return self._projection_belongs_test_data(
+                self.space_args_list, self.shape_list, self.n_points_list
+            )
+
+        def to_tangent_is_tangent_test_data(self):
+            return self._to_tangent_is_tangent_test_data(
+                DiscreteCurves,
+                self.space_args_list,
+                self.shape_list,
+                self.n_vecs_list,
+            )
+
+        def random_tangent_vec_is_tangent_test_data(self):
+            return self._random_tangent_vec_is_tangent_test_data(
+                DiscreteCurves, self.space_args_list, self.n_vecs_list
+            )
+
+        def projection_closed_curves_test_data(self):
+            cells, _, _ = data_utils.load_cells()
+            curves = [cell[:-10] for cell in cells[:5]]
+            ambient_manifold = Euclidean(dim=2)
+            smoke_data = []
+            for curve in curves:
+                smoke_data += [dict(ambient_manifold=ambient_manifold, curve=curve)]
+
+            return self.generate_tests(smoke_data)
+
+    testing_data = DiscreteCurvesTestData()
+
+    @geomstats.tests.np_and_autograd_only
+    def test_projection_closed_curves(self, ambient_manifold, curve):
+        planar_closed_curve = ClosedDiscreteCurves(ambient_manifold)
+        proj = planar_closed_curve.project(curve)
+        expected = proj
+        result = planar_closed_curve.project(proj)
+        self.assertAllClose(result, expected)
+
+        result = proj[-1, :]
+        expected = proj[0, :]
+        self.assertAllClose(result, expected, rtol=10 * gs.rtol)
+
+
+class TestClosedSRVMetric(RiemannianMetricTestCase, metaclass=Parametrizer):
```

Comment:
Nothing works? Were these tested in the current version of our tests, or are these new tests that happen to not run?

Reply:
yup.. this was not tested at all before. These are new tests.